### PR TITLE
Adding module for working with SQS.

### PR DESCRIPTION
**Pull request by maniarathi**

### CODE REVIEW

File: dcplib/aws/sqs_handler.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    def __init__(self, queue_url):
+        self.sqs = boto3.resource('sqs')
+        self.queue_url = queue_url
+
+    def add_message_to_queue(self, payload):
+        """ Given a payload, add it to the queue. Does not return anything, but will raise an exception if the
+        addition was not successful."""
+
+        response = self.sqs.meta.client.send_message(QueueUrl=self.queue_url,
```
Comment (kislyuk) [inline dcplib/aws/sqs_handler.py]: Why not use the resource API and the lazy caching loaders in this directory? Partial sketch:
```python
from . import resources, clients

def __init__(self, queue_name=None, queue_url=None):
    if queue_name:
        self.queue = resources.sqs.Queue(clients.sqs.get_queue_url(QueueName=queue_name))
    else:
        self.queue = resources.sqs.Queue(queue_url)

def send(self, msg, **attributes):
    res = self.queue.send_message(MessageBody=json.dumps(msg), MessageAttributes={k: {"StringValue": v} for k, v in attributes.items()})
    return res['MessageId']
```

Comment (maniarathi) [reply dcplib/aws/sqs_handler.py]: Done.

File: dcplib/aws/sqs_handler.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+class SQSHandler:
+    """
+    A class encapsulating the behaviors associated with interacting with an SQS Queue object and raising errors when
+    queue-related behaviors fail.
+    """
+
+    def __init__(self, queue_name=None, queue_url=None):
+        if queue_name:
+            self.queue = resources.sqs.Queue(clients.sqs.get_queue_url(QueueName=queue_name)['QueueUrl'])
+        else:
```
Comment (kislyuk) [inline dcplib/aws/sqs_handler.py]: Consider this form:
```
if queue_name:
    ...
elif queue_url:
    ...
else:
    raise Exception("Expected either queue_name or queue_url to be specified")
```

File: dcplib/aws/sqs_handler.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+            self.queue = resources.sqs.Queue(queue_url)
+
+    def add_message_to_queue(self, payload, **attributes):
+        """ Given a payload (a dict), add it to the queue. Does not return anything, but will raise an exception if the
```
Comment (kislyuk) [inline dcplib/aws/sqs_handler.py]: > Does not return anything

technically no longer correct :)

File: requirements.txt [hunk 4]
```diff
@@ -1,5 +1,6 @@
 # media_types dependencies
 boto3 >= 1.7.13
 crc32c
+moto
```
Comment (kislyuk) [inline requirements.txt]: This should be in `requirements-dev.txt`, not `requirements.txt`, no?

File: dcplib/aws/sqs_handler.py (hunk 2 above)
Comment (maniarathi) [reply dcplib/aws/sqs_handler.py]: Done

File: dcplib/aws/sqs_handler.py (hunk 3 above)
Comment (maniarathi) [reply dcplib/aws/sqs_handler.py]: Done

File: requirements.txt (hunk 4 above)
Comment (maniarathi) [reply requirements.txt]: Done

**Pull request merged**